### PR TITLE
Mesh wifi issue/ashtam

### PR DESCRIPTION
**Pull request by ashtam55**
Some Minor changes

1. Switch states persistance.

2. RTC memory persistance.

3. MQTT Logging enable.

4. Blink routine added

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -1134,14 +1219,9 @@ void setup()
...
-  if (mqttFlag == false)
-  {
-    initMesh();
-  }
-  else if (mqttFlag == true)
-  {
-    initMeshWithMqtt();
+  initMesh();
+  if (mqttFlag) {
+    updateNodeToMQTTRootNode();
```
Comment (karx) [inline src/main.cpp]: this function `updateNodeToMQTTRootNode` to be called on all possible nodes after providing them WIFI Cred over Mesh or BLE.

BLE is implemented
Also need to publish on Mesh.

File: src/main.cpp [hunk 2]
```diff
@@ -140,6 +148,48 @@ int reset_pin = 23;
...
+    isBlinking = false;
+  }
+  else {
+    blinker.attach(blinkerPace, blink);
+    isBlinking = true;
+  }
+  digitalWrite(STATUS_LED3, LOW);  //make sure LED on on after toggling (pin LOW = led ON)
+}
+
 void pState(byte value)
```
[the remark this answers is not in the document]
Comment (ashtam55) [reply src/main.cpp]: changed to 
```
void isManualResetTriggered(byte value)
```

File: src/main.cpp (hunk 1 above)
Comment (ashtam55) [reply src/main.cpp]: added wifi saving routine in
```
_handlerforMeshAPI(); 
```

File: lib/kaaroLED/kaaroLED.cpp [hunk 3]
```diff
@@ -18,6 +19,7 @@ class KaaroLED
     static int updateState(int switchID, int switchState);
     static boolean* getState();
 
+    static int LastUpdatedState();
```
[the remark this answers is not in the document]
Comment (ashtam55) [reply lib/kaaroLED/kaaroLED.cpp]: changed to 
```
loadLastStateToMem();
```

File: src/main.cpp [hunk 4]
```diff
@@ -15,12 +15,13 @@
...
 #include "kaaroLED.cpp"
 #include "Preferences.h"
+#include <Ticker.h>
 
 /* # FUNCTION PROTOTYPES */
 void mqttCallback(char *topic, byte *payload, unsigned int length);
 IPAddress getlocalIP();
 IPAddress myIP(0, 0, 0, 0);
-
+IPAddress local_IP(192, 168, 0, 115);
```
[the remark this answers is not in the document]
Comment (ashtam55) [reply src/main.cpp]: removed

File: src/main.cpp [hunk 5]
```diff
@@ -15,12 +15,13 @@
 #include "kaaroRTC.cpp"
 #include "kaaroLED.cpp"
 #include "Preferences.h"
+#include <Ticker.h>
```
[the remark this answers is not in the document]
Comment (ashtam55) [reply src/main.cpp]: Ticker is used for heartbeat led - ```STATUS_LED3```

File: lib/kaaroLED/kaaroLED.cpp [hunk 6]
```diff
@@ -2,6 +2,7 @@
 #define _kaaro_led_h_
 
 #include <Arduino.h>
+#include<ArduinoNvs.h>
```
[the remark this answers is not in the document]
Comment (ashtam55) [reply lib/kaaroLED/kaaroLED.cpp]: Both are based on non-volatile storage [Library](https://docs.espressif.com/projects/esp-idf/en/latest/api-reference/storage/nvs_flash.html) classes for esp32 architecture.

File: lib/kaaroLED/kaaroLED.cpp [hunk 7]
```diff
@@ -45,24 +47,37 @@ int KaaroLED::setOutPinConfig(int *OutPinArray, int outPinCount)
...
-        LED_STATE[i] = 0;
+        // LED_STATE[i] = 0;
         pinMode(OUT_PINS[i], OUTPUT);
     }
     out_pin_count = outPinCount;
     return 1;
 }
 
+int KaaroLED::LastUpdatedState(){
+    NVS.begin();
```
[the remark this answers is not in the document]
Comment (ashtam55) [reply lib/kaaroLED/kaaroLED.cpp]: static class ```nvs_esp32``` created for all NVS operations

File: lib/kaaroRTC/kaaroRTC.cpp [hunk 8]
```diff
@@ -4,7 +4,7 @@
 #include "RTClib.h"
 #include <elapsedMillis.h>
 #include "kaaroLED.cpp"
-
+#include <ArduinoNvs.h>
```
[the remark this answers is not in the document]
Comment (ashtam55) [reply lib/kaaroRTC/kaaroRTC.cpp]: added as ```nvs_esp.cpp``

File: src/main.cpp (hunk 1 above)
Comment (karx) [reply src/main.cpp]: Refactoring is good. 
Yet to make all devices connected to internet `choote hi`

**Comment by karx** [comment]
Patch: January01-2020

**Pull request merged**